tokio/src/io/util/async_read_ext.rs

```diff
@@ -181,8 +181,9 @@ cfg_io_util! {
         /// # Return
         ///
         /// On a successful read, the number of read bytes is returned. If the
-        /// supplied buffer is not empty and the function returns `Ok(0)` then
-        /// the source has reached an "end-of-file" event.
+        /// remaining capacity of the supplied buffer is greater than 0 and
+        /// the function returns `Ok(0)` then the source has reached an
+        /// "end-of-file" event.
```

Comment:
I think it would read even better if we made a list with the two cases like we do on [`AsyncReadExt::read`](https://docs.rs/tokio/1.6.1/tokio/io/trait.AsyncReadExt.html#method.read).

Reply:
:+1: I pulled the comment down from read, but changed the second point for `read_buf` because I think the concept of "remaining capacity" makes more sense for `BufMut` than length since a `BufMut` has the concept of initialized length and uninitialized remaining capacity.

Here's the change for that line:
```
- 2. The buffer specified was 0 bytes in length.
+ 2. The buffer specified had a remaining capacity of zero.
```